### ISSUE TITLE
Cannot generate SAS URI per documentation

### ISSUE THREAD

**Issue by stderr-to-devnull** [Azure/azure-cli#17149]
## Describe the bug

**Command Name**
`az storage container generate-sas`

**Errors:**
```
Incorrect padding
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 233, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 664, in execute
    raise ex
cli/core/commands/__init__.py, ln 727, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 719, in _run_job
    return cmd_copy.exception_handler(ex)
...
storage/v2018_11_09/common/_common_conversion.py, ln 91, in _sign_string
    key = _decode_base64_to_bytes(key)
storage/v2018_11_09/common/_common_conversion.py, ln 81, in _decode_base64_to_bytes
    return base64.b64decode(data)
az/lib/python3.6/base64.py, ln 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az storage container generate-sas --connection-string "DefaultEndpointsProtocol=https;AccountName=<used Azure Principal username>;AccountKey=<user Azure Principal user secret>;EndpointSuffix=core.windows.net" --name <an arbitrary container name> --permissions rl --start '2021-02-28T00:00:00Z' --expiry '2021-03-28T00:00:00Z'`

## Environment Summary
```
Linux-5.4.0-66-generic-x86_64-with-debian-bullseye-sid
Python 3.6.10
Installer: DEB

azure-cli 2.19.1

```

**Comment by yonzhan** [comment]
storage

**Comment by stderr-to-devnull** [comment]
Solved the issue by searching the internet and finding an answer on stackoverflow: I needed to use the storage account ID and the storage account ID key which is base64 encoded. The mentioned documentation is the problem and should be adjusted for clarity and also tell the users how to list available storage accounts then how to get the storage account key via azure-cli.

Concerning azure-cli, the storage account key field should parsed internally to check if the key format is base64 to begin with. It is completely different to output an error referncing an incorrect storage account key (and how/where to get it) as opposed to the current situation where a generic python traceback error output is generated leaving people wondering what it could mean. I am definitely not the first person that experienced this issue.
